doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+        * [3.1.1.2 Development Process](#3112-development-process)
+      * [3.1.2 Event Consumer](#312-event-consumer)
+        * [3.1.2.1 Severity](#3121-severity)
+        * [3.1.2.2 Sequence-ID](#3122-sequence-id)
+      * [3.1.3 Alarm Consumer](#313-alarm-consumer)
+      * [3.1.4 Event Receivers](#314-event-receivers)
+        * [3.1.4.1 syslog](#3141-syslog)
+        * [3.1.4.2 REST](#3142-rest)
+        * [3.1.4.3 gNMI](#3143-gnmi)
+        * [3.1.4.4 System LED](#3144-system-led)
+        * [3.1.4.5 Event/Alarm flooding](#3145-event/alarm-flooding)
+        * [3.1.4.6 Eventd continuous restart](#3146-event-continuous-restart)
+      * [3.1.5 Event Profile](#315-event-profile)
+      * [3.1.6 CLI](#316-cli)
+      * [3.1.7 Event History Table and Current Alarm Table](#317-event-history-table-and-current-alarm-table)
+      * [3.1.8 Pull Model](#318-pull-model)
+      * [3.1.9 Supporting third party containers](#319-supporting-third-party-containers)
+    * [3.2 DB Changes](#32-db-changes)
+      * [3.2.1 EVENT DB](#321-event-db)
+    * [3.3 User Interface](#33-user-interface)
+      * [3.3.1 Data Models](#331-data-models)
+      * [3.3.2 CLI](#332-cli)
+        * [3.3.2.1 Exec Commands](#3321-exec-commands)
+        * [3.3.2.2 Configuration Commands](#3322-configuration-commands)
+        * [3.3.2.3 Show Commands](#3323-show-commands)
+      * [3.3.3 REST API Support](#333-rest-api-support)
+  * [4 Flow Diagrams](#4-flow-diagrams)
+  * [5 Warm Boot Support](#5-warm-boot-support)
+    * [5.1 Application warm boot](#51-application-warm-boot)
+    * [5.2 eventd warm boot](#52-eventd-warm-boot)
+  * [6 Scalability](#6-scalability)
+  * [7 Showtech Support](#7-showtech-support)
+  * [8 Unit Test](#8-unit-test)
+
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                                       |
+|:---:|:-----------:|:------------------:|-----------------------------------                       |
+| 0.1 | 03/20/2021  | Srinadh Penugonda  | Initial Version                                          |
+| 0.2 | 04/30/2021  | Srinadh Penugonda  | Updated with comments from HLD review                    |
+
+# About this Manual
+This document provides general information on the implementation and functionality of Event and Alarm Framework in SONiC.
+
+Note: Wherever CLI is specified, it is the CLISH cli that is referred - SONiC native (CLICK) CLI is not updated for this feature.
+
+# Scope
+This document describes the high-level design of Event and Alarm Framework.
+It is not in the scope of the framework to update ANY of the applications to raise events and alarms.
+
+# 1 Feature Overview
+
+The Event and Alarm Framework feature provides a centralized framework for applications in SONiC to raise notifications and store them for NBIs to listen and fetch to monitor the device.
+
+Events and Alarms are means to indicate a change in the state of the system that operator may be interested in. 
```

Comment:
`are means` ---> `means`?

Reply:
Do we need two separate tables: `Event` and `Alarm`? Do we consider alarm is a kind of abnormal event?

Reply:
Updated.
Yes. As mentioned, events are one shot notifications and alarms are dynamic in nature. Alarm table contains those conditions that happened and either need operator attention or may get cleared by those applications. As it is dynamic, and as those conditions gets cleared on a reboot, we need one table to list those alarms. This ultimately can decide system state and system LED.